AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt

```diff
@@ -295,6 +295,9 @@ class CustomStudyDialog : AnalyticsDialogFragment() {
                     setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
                 }
         }
+
+        val canChooseTags = deferredDefaults.getCompleted().tags.isNotEmpty() && contextMenuOption == STUDY_TAGS
```

Comment:
@david-allison I remember you were just in here with something related to deferredDefaults - and when I look at similar code in other areas of this file, I see some care taken with checking `isCompleted`, having a default, or doing a `val defaults = withProgress { deferredDefaults.await() }` etc or a continuation / UI refresh

Seems like that might be necessary here, like perhaps 
```suggestion
        val canChooseTags = (deferredDefaults.isCompleted || deferredDefaults.getCompleted().tags.isNotEmpty()) && contextMenuOption == STUDY_TAGS
```

...related, if it's not completed, is `deferredDefaults.getCompleted()` (or it's `.tags` property) possibly null? 🤔

Reply:
Hi @mikehardy, thanks for the review!

I believe deferredDefaults should be loaded before showing the [custom study main menu](https://github.com/ankidroid/Anki-Android/pull/18562/files/c108ada45c489c71a39079d88f313fa63ac8a171#diff-db6bbcabbd676cdc4c30bd6ff19fe6371503423f866743aeb5ec4cdab543a243L172-L176) loaded.

I like the approach you suggested — it’s more defensive, and I’ll go ahead and add that. Thanks again!